### PR TITLE
Reset Password System

### LINKED ISSUE #39: Forgot/Reset Password System
**Is your feature request related to a problem? Please describe.**
I'm always frustrated when a fellow Borum user forgets his or her password. It pains me to see him or her create a new account because he or she cannot just reset his or her password. There is currently a link in the Login page, but it does not work. In version 0.13.1, please fix all prominent bugs related to the resetting of a user's password. 

**Describe the solution you'd like**
Please complete the functionality for resetting one's password through email. The user should get an email, click the link, fill out their new password, and then immediately be able to log in. 

**Describe alternatives you've considered**
I have not considered any alternatives. 

**Additional context**
ResetPassword1

is the page I am talking about

### PR DESCRIPTION
**Pull request by ghost**
## Overview

**Closes #39**

## Description

Functionality when completed, as follows: 

1. Check if a form using the post method has been submitted
2. Check for validity of the email, and if valid, send the email
3. Whether there is a query string in the url, display the corresponding form
4. Display the footer
Flow of Reset Password System

## Release notes

Notes:

- Working reset password system
 - Use the same email and keep your questions
 - Lets you make a new password

### CODE REVIEW

File: pages/includes/login_functions.inc.php [hunk 1]
```diff
@@ -65,14 +65,14 @@ function check_login($dbc, $email = '', $pass = '') {
...
+function redirect_js($loc, $tim = 1000) {
 	echo "
 	<script>	
 		setTimeout(
 			function() {
 				console.log(\"Redirection timer on\");";
 	echo "window.location.href = '" . $loc . "';";
 	echo "	}
-		, 1000);
+		, $tim);
```
Comment (varunsingh87) [inline pages/includes/login_functions.inc.php:76]: I'm not sure why this is happening now. Please review the entire code throughout the website to see where else this was called.

File: pages/reset_password.php [hunk 2]
```diff
@@ -3,17 +3,41 @@
...
-		echo '<div>An email was sent to your account</div>';	
-		sendEmail('Reset your Password', 'Click the link below to reset your Borum password. ', $e, 'An email was sent to you. Click the link in the email to reset your password.');
+		
+		echo '<div class = "col-sm-10" id = "reset-password-body">';
+		$q = "SELECT SHA2(rand(5000), 512)";
+		$r = mysqli_query($dbc, $q);
+		$row = mysqli_fetch_array($r, MYSQLI_NUM);
+		$q = "INSERT INTO `password-resets` (code, email) VALUES ('{$row[0]}', '$e')";
+		$r = mysqli_query($dbc, $q);
+		sendEmail('Reset your Password', 'Click the link below to reset your Borum password. <br><a href = "http://www.bforborum.com/pages/change_password.php?code=' . $row[0] . '&email=' . $e . '">Click here</a>', $e, 'An email was sent to you. Click the link in the email to reset your password.');
```
Comment (varunsingh87) [inline pages/reset_password.php:31]: Excellent! It is concise, seems to work, and looks consistent with all other mysqli queries.

File: pages/reset_password.php [hunk 3]
```diff
@@ -32,33 +56,9 @@
...
-	</form>
-
-	<?php
-	echo ob_get_clean();
-	exit();
-}
-
 ?>
 
+<!-- If the email has not been identified -->
```
Comment (varunsingh87) [inline pages/reset_password.php:61]: Not bad. First HTML comment of the year.

**Review by varunsingh87 — CHANGES_REQUESTED** [review]

File: pages/includes/login_functions.inc.php (hunk 1 above)
Comment (ghost) [reply pages/includes/login_functions.inc.php:76]: Hi, I did a global search and here’s what I found:

- there were only two files where the parameter was set to a number other than 1000 (including this one above)
- All times that the function was called, it gives the $tim argument a value. 

I guess the development team just didn’t notice the times were all the same?

Comment (varunsingh87) [reply pages/includes/login_functions.inc.php:76]: You're right. Thanks for the response and info. I am merging the commits now.

**Comment by varunsingh87** [comment]
Squash, rebase, or merge?

**Comment by ghost** [comment]
Rebase please

**Pull request merged**